lib/ansible/modules/network/junos/junos_user.py

```diff
@@ -224,8 +224,7 @@ def map_obj_to_ele(module, want):
                 SubElement(user, 'full-name').text = item['full_name']
 
             if item.get('sshkey'):
-                if 'auth' not in locals():
-                    auth = SubElement(user, 'authentication')
+                auth = SubElement(user, 'authentication')
```

Comment:
This assignment may need to check `item['name'] != 'root'` first?

Reply:
I believe this is not necessary as conditionals for root account are implemented in lines 205 to 213.

Reply:
However without checking, `auth` for root will be overwritten.

Reply:
Was there any resolution to this?  We want to avoid regressions in stable releases so this is important to figure out prior to merging.

Reply:
Conditional from line 205 to 209 handles differentiates between root and all other users.
It sets the proper lxml SubElement for each user in the _for_ loop. Changing the SubElement prevents overriding any data between accounts because it changes configuration point in the XML tree.